google/auth/_default.py

```diff
@@ -105,37 +108,41 @@ def _load_credentials_from_file(filename):
-        from google.oauth2 import credentials
-
-        try:
-            credentials = credentials.Credentials.from_authorized_user_info(info)
-        except ValueError as caught_exc:
-            msg = "Failed to load authorized user credentials from {}".format(filename)
-            new_exc = exceptions.DefaultCredentialsError(msg, caught_exc)
-            six.raise_from(new_exc, caught_exc)
-        # Authorized user credentials do not contain the project ID.
-        _warn_about_problematic_credentials(credentials)
-        return credentials, None
-
-    elif credential_type == _SERVICE_ACCOUNT_TYPE:
-        from google.oauth2 import service_account
-
-        try:
-            credentials = service_account.Credentials.from_service_account_info(info)
-        except ValueError as caught_exc:
-            msg = "Failed to load service account credentials from {}".format(filename)
-            new_exc = exceptions.DefaultCredentialsError(msg, caught_exc)
-            six.raise_from(new_exc, caught_exc)
-        return credentials, info.get("project_id")
+    if credential_type is not None:
+        if credential_type == _AUTHORIZED_USER_TYPE:
+            from google.oauth2 import credentials
 
-    else:
-        raise exceptions.DefaultCredentialsError(
-            "The file {file} does not have a valid type. "
-            "Type is {type}, expected one of {valid_types}.".format(
-                file=filename, type=credential_type, valid_types=_VALID_TYPES
-            )
+            try:
+                credentials = credentials.Credentials.from_authorized_user_info(
+                    info, scopes=scopes
+                )
+            except ValueError as caught_exc:
+                msg = "Failed to load authorized user credentials from {}".format(filename)
+                new_exc = exceptions.DefaultCredentialsError(msg, caught_exc)
+                six.raise_from(new_exc, caught_exc)
+            # Authorized user credentials do not contain the project ID.
+            _warn_about_problematic_credentials(credentials)
+            return credentials, None
+
+        elif credential_type == _SERVICE_ACCOUNT_TYPE:
+            from google.oauth2 import service_account
+
+            try:
+                credentials = service_account.Credentials.from_service_account_info(
+                    info, scopes=scopes
+                )
+            except ValueError as caught_exc:
+                msg = "Failed to load service account credentials from {}".format(filename)
+                new_exc = exceptions.DefaultCredentialsError(msg, caught_exc)
+                six.raise_from(new_exc, caught_exc)
+            return credentials, info.get("project_id")
+
+    raise exceptions.DefaultCredentialsError(
+        "The file {file} does not have a valid type. "
+        "Type is {type}, expected one of {valid_types}.".format(
+            file=filename, type=credential_type, valid_types=_VALID_TYPES
         )
+    )
```

Comment:
The control flow is a little tricky to eyeball grep. It may be easier if it were changed to be something like
```python
if credential_type is None:
    raise exceptions.DefaultCredentialsError(...)
# dedented control flow
```